leaderboard/redis/cluster_client.go

```diff
@@ -44,33 +45,42 @@ func (cc *clusterClient) ExpireAt(ctx context.Context, key string, time time.Tim
 // Ping call redis PING function
 func (cc *clusterClient) Ping(ctx context.Context) error {
 	err := cc.ClusterClient.Ping(ctx).Err()
-	return err
+	if err != nil {
+		return NewUnknownError(err.Error())
+	}
+	return nil
 }
 
 // SAdd call redis SADD function
 func (cc *clusterClient) SAdd(ctx context.Context, key, member string) error {
-	_, err := cc.ClusterClient.SAdd(ctx, key, member).Result()
-	return err
+	err := cc.ClusterClient.SAdd(ctx, key, member).Err()
+	if err != nil {
+		return NewUnknownError(err.Error())
+	}
+	return nil
 }
 
 // SRem call redis SREM function
 func (cc *clusterClient) SRem(ctx context.Context, key, member string) error {
 	err := cc.ClusterClient.SRem(ctx, key, member).Err()
-	return err
+	if err != nil {
+		return NewUnknownError(err.Error())
+	}
+	return nil
 }
 
 // TTL call redis TTL function
 func (cc *clusterClient) TTL(ctx context.Context, key string) (time.Duration, error) {
 	result, err := cc.ClusterClient.TTL(ctx, key).Result()
 	if err != nil {
-		return -1, err
+		return -1, NewUnknownError(err.Error())
```

Comment:
Won't we lose the error details if we do that? Can we access the stack trace when we wrap an error like that?